File: project-apply/backend/src/infrastructure/external/scraper/src/domain/entities/candidate.py
from dataclasses import dataclass
from datetime import datetime
from typing import Optional


@dataclass
class Candidate:
    id: str
    name: str
    email: str
    phone: Optional[str]
    location: Optional[str]
    skills: list[str]
    experience_years: int
    title: str
    summary: Optional[str]
    resume_url: Optional[str]
    created_at: datetime
    updated_at: datetime

    def to_dict(self) -> dict:
        return {
            "id": self.id,
            "name": self.name,
            "email": self.email,
            "phone": self.phone,
            "location": self.location,
            "skills": self.skills,
            "experience_years": self.experience_years,
            "title": self.title,
            "summary": self.summary,
            "resume_url": self.resume_url,
            "created_at": self.created_at.isoformat(),
            "updated_at": self.updated_at.isoformat(),
        }

File: project-apply/backend/src/domain/entities/Candidate.ts
export enum CandidateStatus {
  ACTIVE = 'active',
  INACTIVE = 'inactive',
  HIRED = 'hired',
  WITHDRAWN = 'withdrawn',
}

export interface Candidate {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  phone?: string;
  location?: string;
  linkedinUrl?: string;
  resumePath?: string;
  skills: string[];
  experienceYears?: number;
  currentCompany?: string;
  currentTitle?: string;
  status: CandidateStatus;
  consentGdpr: boolean;
  createdBy?: string;
  createdAt: Date;
  updatedAt: Date;
}

export interface CreateCandidateDTO {
  firstName: string;
  lastName: string;
  email: string;
  phone?: string;
  location?: string;
  linkedinUrl?: string;
  skills?: string[];
  experienceYears?: number;
  currentCompany?: string;
  currentTitle?: string;
  consentGdpr?: boolean;
}

export interface UpdateCandidateDTO {
  firstName?: string;
  lastName?: string;
  email?: string;
  phone?: string;
  location?: string;
  linkedinUrl?: string;
  skills?: string[];
  experienceYears?: number;
  currentCompany?: string;
  currentTitle?: string;
  status?: CandidateStatus;
  consentGdpr?: boolean;
}
